pkg/mcp/source/client_source.go

```diff
@@ -51,6 +54,35 @@ func NewClient(client mcp.ResourceSinkClient, options *Options) *Client {
 
 var reconnectTestProbe = func() {}
 
+// Some scenarios requires the client to send the first message in a
```

Comment:
Follow-up: we need to document this convention in the MCP specification under istio/api.

Reply:
Also, we might need to update the logging on the server (sink) side to make this NACK less alarming to the user/dashboard.